synapse/handlers/sliding_sync/__init__.py

```diff
@@ -1242,3 +1277,167 @@ async def _get_bump_stamp(
                     return new_bump_event_pos.stream
 
             return None
+
+
+def _required_state_changes(
+    user_id: str,
+    previous_room_config: "RoomSyncConfig",
```

Comment:
Since these are the same type, it would be good to require keyword args so people don't accidentally mess up the order.

```suggestion
    *,
    previous_room_config: "RoomSyncConfig",
```